### PR TITLE
sstp: T2566: Fix to allow IPv6 only pools

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

To allow IPv6 only for vpn sstp sessions we have to add
'ppp-options' which can disable IPv4 allocation explicitly.
Additional IPv6 ppp-options and fix template for it.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T2566

## Component(s) name

sstp
## How to test
VyOS sstp configuration:
```
set vpn sstp authentication local-users username foo password 'bar'
set vpn sstp authentication local-users username foo2 password 'bar2'
set vpn sstp authentication mode 'local'
set vpn sstp client-ipv6-pool prefix 2001:db8::/48
set vpn sstp gateway-address '192.168.122.14'
set vpn sstp ppp-options ipv4 'deny'
set vpn sstp ppp-options ipv6 'allow'
set vpn sstp ssl ca-cert-file '/config/user-data/sstp/ca.crt'
set vpn sstp ssl cert-file '/config/user-data/sstp/server.crt'
set vpn sstp ssl key-file '/config/user-data/sstp/server.key'
```
Check sessions:
```
vyos@r4-epa2# run show sstp-server sessions 
ifname | username |          ip           |          ip6          | ip6-dp |   calling-sid   | rate-limit | state  |  uptime  | rx-bytes | tx-bytes 
--------+----------+-----------------------+-----------------------+--------+-----------------+------------+--------+----------+----------+----------
 sstp0  | foo2     | 2001:db8:0:0:200::/64 | 2001:db8:0:0:200::/64 |        | 192.168.122.222 |            | active | 00:12:45 | 2.9 KiB  | 946 B
[edit]
vyos@r4-epa2# 

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: data/templates/accel-ppp/config_ipv6_pool.j2 [hunk 1]
```diff
@@ -5,7 +5,7 @@ AdvAutonomousFlag=1
 {%   if client_ipv6_pool.prefix is defined and client_ipv6_pool.prefix is not none %}
 [ipv6-pool]
 {%     for prefix, options in client_ipv6_pool.prefix.items() %}
-{{ prefix }},{{ options.mask }}
+{{ prefix }},{{ '64 ' if options.mask is not defined else options.mask }}
```
Comment (c-po) [inline data/templates/accel-ppp/config_ipv6_pool.j2:8]: Why not specify `<defaultValue>64</defaultValue>` for mask in XML? The no special treatment is required here.

SSTP already uses `get_accel_dict()` which blends in the default values automatically in XML.

**Comment by c-po** [comment]
Any reason this PR is for `equuleus` and not for `current`?

**Review by c-po — CHANGES_REQUESTED** [review]

File: data/templates/accel-ppp/config_ipv6_pool.j2 (hunk 1 above)
Comment (sever-sever) [reply data/templates/accel-ppp/config_ipv6_pool.j2:8]: I tried it, but it had incorrect behavior
It starts to see "mask" as "prefix"
For example when I use "defaultValue" 64 in XML
and such template:
```
{%     for prefix, options in client_ipv6_pool.prefix.items() %}
{{ prefix }},{{ options.mask }}
{%     endfor %}

```
It generates the next config (if the `mask` was not added in configuration):
`set vpn sstp client-ipv6-pool prefix 2001:db8::/48`
```
[ipv6-pool]
2001:db8::/48,
mask,
```
It is incorrect, I expect prefix /64 by default
**Expected configuration:**
```
[ipv6-pool]
2001:db8::/48,64
```
If I add mask in configuration:
`set vpn sstp client-ipv6-pool prefix 2001:db8::/48 mask 64`
it generates:
```
[ipv6-pool]
2001:db8::/48,64
mask,

```

**Comment by c-po** [comment]
Yeah its a nesting problem - will resolve it myself and link the commit here.

**Comment by c-po** [comment]
https://github.com/vyos/vyos-1x/commit/01ed77040ec9493e4ca1cf868ff3c22847da4487

**Pull request merged**